Scripts/Runtime/Entities/TaskDriver/TaskSet/TaskData/DataStream/DataSource/Consolidator/CancelRequestsDataSourceConsolidator.cs

```diff
@@ -0,0 +1,61 @@
+using Anvil.Unity.DOTS.Data;
+using Anvil.Unity.DOTS.Jobs;
+using System;
+using System.Collections.Generic;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+
+namespace Anvil.Unity.DOTS.Entities.Tasks
+{
+    [BurstCompatible]
+    internal struct CancelRequestsDataSourceConsolidator : IDisposable
+    {
+        private const int UNSET_THREAD_INDEX = -1;
+
+        [NativeSetThreadIndex] [ReadOnly] private readonly int m_NativeThreadIndex;
+        
+        private UnsafeTypedStream<EntityProxyInstanceID> m_Pending;
+        private UnsafeParallelHashMap<uint, CancelRequestsActiveConsolidator> m_ActiveConsolidatorsByID;
+
+        public CancelRequestsDataSourceConsolidator(PendingData<EntityProxyInstanceID> pendingData, Dictionary<uint, AbstractData> dataMapping)
```

Comment:
🚲 Not really worth changing now but something to keep in mind. We can pass in `IReadOnlyDictionary` to better communicate what the type will be doing with this dictionary

Reply:
Fair point